core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/V4HttpSigner.java

```diff
@@ -100,9 +100,9 @@ public AsyncSignedRequest signAsync(AsyncSignRequest<? extends AwsCredentialsIde
 
         Publisher<ByteBuffer> payload = payloadSigner.sign(request.payload().orElse(null), futureV4Context);
```

Comment:
I was going to say this could be moved inside the thenApply below, but see you made that change in https://github.com/aws/aws-sdk-java-v2/pull/4270/files#diff-e7528bac3b3393210083f8bf3ef4b519b2d5754e7771d9aab68d6a8b82848dc4. I guess it could have be brought into this PR, but ok that it comes in the next PR.